packages/astro/src/core/create-vite.ts

```diff
@@ -239,6 +239,8 @@ export async function createVite(
 	// We also need to filter out the plugins that are not meant to be applied to the current command:
 	// - If the command is `build`, we filter out the plugins that are meant to be applied for `serve`.
 	// - If the command is `dev`, we filter out the plugins that are meant to be applied for `build`.
+	const assetsDir = settings.config.build.assets;
+	if (assetsDir) result = vite.mergeConfig(result, { build: { assetsDir } });
```

Comment:
Is there a reason why we only set when `build.assets` is specified? Shouldn't we always pass it?